src/component/legend/ScrollableLegendView.js

```diff
@@ -188,44 +206,85 @@ var ScrollableLegendView = LegendView.extend({
             var pageButtonPosition = legendModel.get('pageButtonPosition', true);
             // controller is on the right / bottom.
             if (pageButtonPosition === 'end') {
-                controllerPos[orientIdx] += maxSize[wh] - controllerRect[wh];
+                if (selectorGroup) {
+                    if (selectorPosition === 'end') {
+                        selectorPos[orientIdx] += maxSize[wh] - selectorRect[wh];
+                        controllerPos[orientIdx] += maxSize[wh] - selectorRect[wh]
+                            - selectorLegendGap - controllerRect[wh];
+                    }
+                    else {
+                        controllerPos[orientIdx] += maxSize[wh] - controllerRect[wh];
+                        containerPos[orientIdx] += selectorRect[wh] + selectorLegendGap;
+                    }
+                }
+                else {
+                    controllerPos[orientIdx] += maxSize[wh] - controllerRect[wh];
+                }
             }
             // controller is on the left / top.
             else {
-                containerPos[orientIdx] += controllerRect[wh] + pageButtonGap;
+                if (selectorGroup) {
+                    if (selectorPosition === 'start') {
+                        controllerPos[orientIdx] += selectorRect[wh] + selectorLegendGap;
+                        containerPos[orientIdx] += selectorRect[wh] + selectorLegendGap
+                            + controllerRect[wh] + pageButtonGap;
+                    }
+                    else {
+                        selectorPos[orientIdx] += maxSize[wh] - selectorRect[wh];
+                        containerPos[orientIdx] += controllerRect[wh] + pageButtonGap;
+                    }
+                }
+                else {
+                    containerPos[orientIdx] += controllerRect[wh] + pageButtonGap;
+                }
+            }
+        }
+        else if (selectorGroup) {
+            if (selectorPosition === 'end') {
+                selectorPos[orientIdx] += contentRect[wh] + selectorLegendGap;
+            }
+            else {
+                containerPos[orientIdx] += selectorRect[wh] + selectorLegendGap;
```

Comment:
The code is a little complicated I think. We should not modify the containerPos, but layout selector based on them.

That is, we should not modify the original layout logic of container.
The layout of the selector should be after the layout of container.
Do not mix them, which makes the logic complicated and hard to understand.

Reply:
Without selector, the containerPos will also be modified by controller.
You can the code on branck master:
 https://github.com/apache/incubator-echarts/blob/master/src/component/legend/ScrollableLegendView.js#L195